controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -90,58 +147,232 @@ int main(int argc, char** argv) {
-                    if (distance < minDistance) {
-                        collision = true;
+                    // Loop through the vector of existing proposed locations and see if the new one is going to
+                    // collide with any of them
+                    for (const auto& testPos : positions) {
+                        const double distance =
+                            std::sqrt(std::pow((newPos[1] - testPos[1]), 2) + std::pow((newPos[0] - testPos[0]), 2));
+                        if (distance < minDistance) {
+                            collision = true;
+                        }
                     }
-                }
-                // Loop until a proposed location has been found that doesn't clash with the existing ones
-            } while (collision);
-            // Finally add the proposed location in as a confirmed position
-            positions.emplace_back(newPos);
+                    // Loop until a proposed location has been found that doesn't clash with the existing ones
+                } while (collision);
+                // Finally add the proposed location in as a confirmed position
+                positions.emplace_back(newPos);
+            }
+
+            // Loop through every robot
+            for (size_t i = 0; i < robotsNodes.size(); i++) {
+                // Grab translation field of the robot to modify
+                // There will be a position for every robot in the positions vector
+                robotsNodes[i]->getField("translation")->setSFVec3f(positions[i].data());
+                // Apply new rotation
+                robotsNodes[i]->getField("rotation")->setSFRotation(rotations[size_t(rotDistrib(gen))].data());
+
+                // Reset physics to avoid robot tearing itself apart
+                robotsNodes[i]->resetPhysics();
+            }
+            // Set random positions for the head and neck servos
+            const double neck_yaw_position   = servoDistrib(gen);
+            const double head_pitch_position = servoDistrib(gen);
+            neck_yaw->setPosition(neck_yaw_position);
+            head_pitch->setPosition(head_pitch_position);
         }
-        // Loop through every robot
-        for (size_t i = 0; i < otherRobotsNodes.size(); i++) {
-            // Grab translation field of the robot to modify
-            // There will be a position for every robot in the positions vector
-            otherRobotsNodes[i]->getField("translation")->setSFVec3f(positions[i].data());
-            // Apply new rotation
-            otherRobotsNodes[i]->getField("rotation")->setSFRotation(rotations[size_t(rotDistrib(gen))].data());
-
-            // Reset physics to avoid robot tearing itself apart
-            otherRobotsNodes[i]->resetPhysics();
+
+        // Run a number of iterations after the robot has moved equal to the value of modulo
+        if (moduloCounter % modulo == 0) {
```

Comment:
Why not just do inside the while loop
- Move robot
- Call `supervisor.step(timeStep)` (or `timeStep*2` if needed?)
- Save data

Rather than having all the modulo stuff?

Reply:
If that would do the same thing I can change it... It just made more sense in my head to have the while loop run every time rather than stepping halfway through the loop

Reply:
Up to you, they're just different ways to do it - I thought it might be a bit neater but it doesn't really matter.